colossalai/fx/passes/algorithms/ckpt_solver_rotor.py

```diff
@@ -353,10 +353,10 @@ def solver_rotor(gm: ColoGraphModule,
             logger.info("dynamic_programs_C_version hasn't been built! Building library...", ranks=[0])
             this_dir = os.path.dirname(os.path.abspath(__file__))
             result = subprocess.Popen(
-                f'python {os.path.join(this_dir, "build_c_ext.py")} build_ext --build-lib={this_dir}',
+                ["python", f"{os.path.join(this_dir, 'build_c_ext.py')}", "build_ext", f"--build-lib={this_dir}"],
```

Comment:
You should not use `python`. To be safe, you should use `sys.executable` to give the absolute path to the python bin.

Reply:
okay